kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -431,15 +431,18 @@ def _init_term(
     return substitute(empty_config, init_subst)
 
 
-def _final_cterm(empty_config: KInner, contract_name: str, *, failing: bool) -> CTerm:
+def _final_cterm(empty_config: KInner, contract_name: str, *, failing: bool, is_test: bool = True) -> CTerm:
     final_term = _final_term(empty_config, contract_name)
     key_dst = KEVM.loc(KToken('FoundryCheat . Failed', 'ContractAccess'))
     dst_failed_post = KEVM.lookup(KVariable('CHEATCODE_STORAGE_FINAL'), key_dst)
     foundry_success = Foundry.success(KVariable('STATUSCODE_FINAL'), dst_failed_post)
-    if not failing:
-        return CTerm(final_term).add_constraint(mlEqualsTrue(foundry_success))
-    else:
-        return CTerm(final_term).add_constraint(mlEqualsTrue(notBool(foundry_success)))
+    final_cterm = CTerm(final_term)
+    if is_test:
+        if not failing:
+            return final_cterm.add_constraint(mlEqualsTrue(foundry_success))
+        else:
+            return final_cterm.add_constraint(mlEqualsTrue(notBool(foundry_success)))
```

Comment:
Why do we need to add a constraint on the `foundry_success` predicate here?

Reply:
Here we are adding, to the `final_cterm`, the constraint `{ true #Equals foundry_success(...) }`. We're not adding a constarint to `foundry_success`, we're adding the `foundry_success` constraint to the CTerm.

Reply:
You can find the implementation of `add_constraint` here: https://github.com/runtimeverification/pyk/blob/6bb1fcb27ddfa0d9c9bd45699951ad53ad9d678d/src/pyk/cterm.py#L95

Reply:
That predicate is only added for `test*` functions, see the outer `if ...` statement.

Reply:
The one where it says `if is_test: ...`